packages/react-heartwood-components/src/components/View/View.js

```diff
@@ -11,40 +11,51 @@ type Props = {
 	business: Object,
 	getSearchSuggestionValue?: Function,
 	renderSearchSuggestion?: Function,
-	children: Node
+	children: Node,
+	toggleSidebarVisibility?: Function,
+	toggleSidebarExpanded?: Function,
+	isSidebarVisible?: boolean,
+	isSidebarExpanded?: boolean
 }
 type State = {
-	sidebarIsVisible: boolean,
-	sidebarIsExpanded: boolean
+	isSidebarVisible: boolean,
+	isSidebarExpanded: boolean
 }
 
 export default class View extends Component<Props, State> {
+	static defaultProps = {
+		isSidebarVisible: false,
+		isSidebarExpanded: true
+	}
+
 	state = {
-		sidebarIsVisible: false,
-		sidebarIsExpanded: true
+		isSidebarVisible: this.props.isSidebarVisible,
+		isSidebarExpanded: this.props.isSidebarExpanded
 	}
 
 	toggleSidebarVisibility = () => {
 		this.setState(prevState => ({
-			sidebarIsVisible: !prevState.sidebarIsVisible,
-			sidebarIsExpanded: true
+			isSidebarVisible: !prevState.isSidebarVisible,
+			isSidebarExpanded: true
 		}))
+		this.props.toggleSidebarVisibility && this.props.toggleSidebarVisibility()
```

Comment:
This isn't exactly a controlled component, since it maintains its own state here. More typical would be to call something like `onSidebarVisibilityChange`, and not keep internal state (only refer to the props, and defer to the downstream components to manage state). 

A challenge might be to both keep local state if `onSidebarVisibilityChange` was not provided, and otherwise use props if it was. So in that way you could have both a controlled and an uncontrolled mode. Necessary? I don't know. We really only *need* a controlled mode for the website.

Reply:
Confirming there is no point in having a stateful component here. Going fully controlled is the way to go!